Assets/MixedRealityToolkit.Services/SpatialAwarenessSystem/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -177,17 +177,17 @@ public IMixedRealitySpatialAwarenessObserver GetObserver(string name)
         }
 
         /// <inheritdoc />
-        public IMixedRealitySpatialAwarenessObserver GetObserver<T>(string name) where T : IMixedRealitySpatialAwarenessObserver
+        public T GetObserver<T>(string name) where T : IMixedRealitySpatialAwarenessObserver
         {
             for (int i = 0; i < observers.Count; i++)
             {
                 if ((observers[i] is T) && (observers[i].Name == name))
                 {
-                    return observers[i];
+                    return (T)observers[i];
                 }
             }
-
-            return null;
```

Comment:
Is the expectation that T implements IMixedRealitySpatialAwarenessObserver? Or will there be asks for object types that don't directly implement this interface but then resolve correctly in the type check?

Reply:
The "where T : IMixedRealitySpatialAwarenessObserver" statement tells the compiler that T must implement IMixedRealitySpatialAwarenessObserver. All observers need to implement this interface, plus any additional interfaces / functionality required.